content/api/commands/origin.md

```diff
@@ -0,0 +1,512 @@
+cy.get('h1').contains('My cool site under test')
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> url** **_(String)_**
+
+A URL specifying the secondary origin in which the callback is to be executed.
+This should at the very least contain a hostname, and may also include the
+protocol, port number & path. Query params are not supported.
+
+This argument will be used in two ways:
+
+1. It uniquely identifies a secondary origin in which the commands in the
+   callback will be executed. Cypress will inject itself into this origin, and
+   then send it code to evaluate in that origin, without violating the browser's
+   same-origin policy.
+
+2. It overrides the `baseUrl` configured in your
+   [global configuration](/guides/references/configuration#Global) while inside
+   the callback. So `cy.visit()` and `cy.request()` will use this URL as a
+   prefix, not the configured `baseUrl`.
+
+**<Icon name="angle-right"></Icon> options** **_(Object)_**
+
+Pass in an options object to control the behavior of `cy.origin()`.
+
+| option | description                                                                                                                                                                                                    |
+| ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| args   | Plain JavaScript object which will be serialized and sent from the primary origin to the secondary origin, where it will be deserialized and passed into the callback function as its first and only argument. |
+
+<Alert type="warning">
+
+The `args` object is the **only** mechanism via which data may be injected into
+the callback, the callback is **not** a closure and does not retain access to
+the JavaScript context in which it was declared. Values passed into `args`
+**must** be serializable.
+
+</Alert>
+
+**<Icon name="angle-right"></Icon> callbackFn** **_(Function)_**
+
+The function containing the commands to be executed in the secondary origin.
+
+This function will be stringified, sent to the Cypress instance in the secondary
+origin and evaluated. If the `args` option is specified, the deserialized args
+object will be passed into the function as its first and only argument.
+
+There are a number of limitations placed on commands run inside the callback,
+please see [Callback restrictions](#Callback-restrictions) section below for a
+full list.
+
+### Yields [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Subject-Management)
+
+- `cy.origin()` yields the value yielded by the last Cypress command in the
+  callback function.
+- If the callback contains no Cypress commands, `cy.origin()` yields the return
+  value of the function.
+- In either of the two cases above, if the value is not serializable,
+  `cy.origin()` yields null.
```

Comment:
@mjhenkes is this correct or do we throw an error?

Reply:
It's not correct. We don't yield null if the subject is not serializable. But we don't throw immediately either. We only throw if the tests tries to access the unserializable value.

So this is fine:

```js
cy.origin('foobar.com', () => {
  cy.get('#foo') // yields element (not serializable)
})
// .. other commands ...
```

This will throw an error:

```js
cy.origin('foobar.com', () => {
  cy.get('#foo') // yields element (not serializable)
})
.then(($el) => {
  $el.text() // <- throws at this point
})
```

Reply:
I'm fixing this